### PR TITLE
Camera-ready version

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
This mostly shrinks the document to fit in 12 pages.
We will need to do a final update once we get the official DOI

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
@ryzhyk you should probably build this to make sure it works, but a careful proofreading may not be necessary if you are busy.

File: doc/vldb23/intro2.tex [hunk 1]
```diff
@@ -95,12 +95,12 @@ \section{Introduction}\label{sec:introduction}
...
-Conversely, the database snapshot at time $t$ is the cumulative result of applying all
-transactions up to $t$: $DB[t] = \sum_{i \leq t} \Delta DB[i]$.
-The operation of adding up all changes is the inverse of differentiation,
-and is another basic stream operator, \emph{stream integration}: $\I$.
-The following diagram expresses the relationship between the streams $\Delta DB$ and $DB$:
-
+Conversely, the database snapshot at time $t$ is the cumulative result
+of applying all transactions up to $t$: $DB[t] = \sum_{i \leq t}
+\Delta DB[i]$.  The operation $\I$, adding up all changes another
+basic stream operator, is \emph{stream integration}, the inverse of
```
Comment (ryzhyk) [inline doc/vldb23/intro2.tex:101]: typo

File: doc/vldb23/intro2.tex [hunk 2]
```diff
@@ -177,12 +176,12 @@ \section{Introduction}\label{sec:introduction}
 maintains the transitive closure of a graph as the graph is changed by
 adding and deleting edges.
 
-Due to space reasons, this paper omits proofs.  A companion extensive
-technical report\anonymize{~\cite{tr}} contains the proofs.  We have
-formalized the entire \dbsp theory in the Lean proof
+In this paper we omits proofs, they can be found in an extensive
```
Comment (ryzhyk) [inline doc/vldb23/intro2.tex:179]: omit

File: doc/vldb23/intro2.tex [hunk 3]
```diff
@@ -177,12 +176,12 @@ \section{Introduction}\label{sec:introduction}
...
+companion technical report\anonymize{~\cite{tr}}.  We have formalized
+this theory in the Lean proof
 assistant~\anonymize{\cite{dbsp-theory}}; our formalization includes
-machine-checked proofs of correctness for all the theorems stated in
-this paper.
+machine-checked proofs of correctness for all the theorems in this
```
Comment (ryzhyk) [inline doc/vldb23/intro2.tex:183]: "of correctness" is redundant here

File: doc/vldb23/relational.tex [hunk 4]
```diff
@@ -621,38 +619,38 @@ \subsection{Complexity of incremental circuits}\label{sec:complexity}
...
+Algorithm~\ref{algorithm-inc}.  The question of optimality in the
+context of IVM plan is a much more difficult topic than optimization
+of ad-hoc queries, since the chosen IVM plan will execute for
+\emph{all future database updates}.
 
 Moreover, since incremental computations maintain internal state, it
 follows that incremental plans cannot be simply changed in-flight,
 like we can change ad-hoc queries based on current data statistics:
-deploying a new requires it in general to reconstruct its internal
+deploying a new plan requires in general to construct its internal
```
Comment (ryzhyk) [inline doc/vldb23/relational.tex:650]: to construct -> constructing

File: doc/vldb23/streams.tex [hunk 5]
```diff
@@ -169,8 +167,8 @@ \subsubsection{Delays and time-invariance}\label{sec:delay}
 
 \begin{definition}[Time invariance]
 A stream operator $S: \stream{A} \to \stream{B}$ is \defined{time-invariant} (TI) if
-$S(\zm_A(s)) = \zm_B(S(s))$ for all $s \in \stream{A}$, or, in other words, if the
-following two circuits are equivalent:
+$S(\zm_A(s)) = \zm_B(S(s))$ for all $s \in \stream{A}$; in other words, the
```
Comment (ryzhyk) [inline doc/vldb23/streams.tex:170]: typo

File: doc/vldb23/streams.tex [hunk 6]
```diff
@@ -321,8 +313,8 @@ \subsection{Integration and differentiation}\label{sec:abelianstreams}
 \end{definition}
 
 This definition extends to stream operators.
-Lifting a bilinear function $f: A \times B \to C$ produces
-a bilinear stream operator $\lift{f}$.  An example bilinear operator over $\stream{\Z}$
+The lifting a bilinear function $f$ is
```
Comment (ryzhyk) [inline doc/vldb23/streams.tex:316]: of

File: doc/vldb23/streams.tex [hunk 7]
```diff
@@ -463,7 +455,7 @@ \subsection{Integration and differentiation}\label{sec:abelianstreams}
 
 \section{Incremental view maintenance}\label{sec:incremental}
 
-In this section we give a precise definition of IVM and analyze its properties.
+Here we define of IVM and analyze its properties.
```
Comment (ryzhyk) [inline doc/vldb23/streams.tex:458]: remove "of"

**Review by ryzhyk — CHANGES_REQUESTED** [review]

**Comment by mbudiu-vmw** [comment]
I will wait for Tej to comment before I commit a new pdf

**Comment by mbudiu-vmw** [comment]
@ryzhyk this is supposed to be the final-final version of the vldb paper, please let me know if I can merge it.

**Review by ryzhyk — APPROVED** [review]

**Pull request merged**